core/modules/layout/includes/block.text.inc

```diff
@@ -90,26 +90,20 @@ class BlockText extends Block {
         $form['convert']['reusable']['#description'] .= ' ' . t('Reusable blocks may also be translated.');
       }
     }
-    $form['admin_label'] = array(
-      '#type' => 'fieldset',
-      '#title' => t('Admin label'),
-      '#tree' => FALSE,
-      '#collapsed' => empty($this->settings['admin_label']),
-      '#states' => array(
-        'collapsed' => array(
-          ':input[name="reusable"]' => array('checked' => FALSE),
-        ),
+
+    // Adjust the admin label fieldset based on block reusability.
+    $form['admin_label']['#weight'] = 4;
+    $form['admin_label']['#collapsible'] = TRUE;
+    $form['admin_label']['#states'] = array(
+      'collapsed' => array(
+        ':input[name="reusable"]' => array('checked' => FALSE),
       ),
-      '#collapsible' => TRUE,
-      '#weight' => 4,
-    );
-    $form['admin_label']['admin_label'] = array(
-      '#type' => 'textfield',
-      '#default_value' => isset($this->settings['admin_label']) ? $this->settings['admin_label'] : '',
-      '#title' => t('Admin label'),
-      '#description' => t('Used to identify the block on layout pages. Required for reusable blocks.'),
-      '#maxlength' => 255,
     );
+
+    // Add additional description text for reusable blocks.
+    $form['admin_label']['admin_label']['#description'] .= t('Required for reusable blocks.');
```

Comment:
Tests are failing here with "Undefined index: #description". This must be because the "Used to identify the block on layout pages." bit that was there in this description before has now been moved to the fieldset description instead. This should fix things:

```suggestion
    $form['admin_label']['admin_label']['#description'] = t('Required for reusable blocks.');
```